docs/ref/templates/builtins.txt

```diff
@@ -2846,6 +2846,10 @@ For example:
 
 If ``value`` is ``"Joel is a slug"``, the output will be ``"JOEL IS A SLUG"``.
 
+.. versionchanged:: 5.2
+
+    In older versions, the ``upper`` output wasn't marked as safe.
```

Comment:
I think the behaviour here is that `upper` should preserve the safeness of the input. I think this wording clarifies that:

```suggestion
    In older versions, the ``upper`` output wasn't marked as safe when the input was safe.
```